perf(napi/parser, napi/oxlint): lazy visit: faster check for exit visitor

overlookmotel opened the pull request:
In the absence of benchmarks, I don't know for sure that this is a perf improvement. But I assume so.

`if (exit)` requires coercion of `exit` to a boolean, whereas `if (exit !== null)` is a simpler check - comparison to a single known constant.

Once we have reliable benchmarking set up for raw transfer, we can revisit and tweak the code in many ways, but I think in the meantime this change is *probably* an improvement.

overlookmotel commented:
* **#12496** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12496?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12496?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12496).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 reviewed (approved):
```
import { Bench } from 'tinybench';

const bench = new Bench({ name: 'simple benchmark', time: 100 });

const data = [];
for (let i = 0; i < 1000; i++) {
    data.push(Math.random() > 0.5 ? () => {} : null);
}

bench
    .add('coerce boolean', () => {
        for (const item of data) {
            if (item) {
                item();
            }
        }
    })
    .add('vs null', async () => {
        for (const item of data) {
            if (item !== null) {
                item();
            }
        }
    });

await bench.run();

console.log(bench.name);
console.table(bench.table());
```

```
┌─────────┬──────────────────┬──────────────────┬──────────────────┬────────────────────────┬────────────────────────┬─────────┐
│ (index) │ Task name        │ Latency avg (ns) │ Latency med (ns) │ Throughput avg (ops/s) │ Throughput med (ops/s) │ Samples │
├─────────┼──────────────────┼──────────────────┼──────────────────┼────────────────────────┼────────────────────────┼─────────┤
│ 0       │ 'coerce boolean' │ '2326.9 ± 0.21%' │ '2292.0 ± 1.00'  │ '431865 ± 0.03%'       │ '436300 ± 190'         │ 42977   │
│ 1       │ 'vs null'        │ '1756.3 ± 0.37%' │ '1750.0 ± 0.00'  │ '572547 ± 0.03%'       │ '571429 ± 0'           │ 56937   │
└─────────┴──────────────────┴──────────────────┴──────────────────┴────────────────────────┴────────────────────────┴─────────┘
```


‘vs null’ is faster than ‘coerce boolean’ by
 - ~24.5% in latency
 - ~32.6% in throughput

camc314 commented:
### Merge activity

* **Jul 24, 10:26 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
perf(napi/parser, napi/oxlint): lazy visit: faster check for exit visitor (#12496)

In the absence of benchmarks, I don't know for sure that this is a perf improvement. But I assume so.

`if (exit)` requires coercion of `exit` to a boolean, whereas `if (exit !== null)` is a simpler check - comparison to a single known constant.

Once we have reliable benchmarking set up for raw transfer, we can revisit and tweak the code in many ways, but I think in the meantime this change is *probably* an improvement.
